core/base-service/auth-helper.js

```diff
@@ -151,6 +151,11 @@ class AuthHelper {
     return this.isConfigured ? { Authorization: `Bearer ${pass}` } : undefined
   }
 
+  get _botAuthHeader() {
+    const { _pass: pass } = this
+    return this.isConfigured ? { Authorization: `Bot ${pass}` } : undefined
```

Comment:
to avoid duplicate code as well as platform specific code in this generic helper, you might consider refactoring this logic to accept the concept of a "token type", and then configure discord to be a `Bot` token type (and you could default it to `Bearer`)

Reply:
That's precisely what was there originally but the engine in LGTM was barking about hard coded credentials because of it

https://github.com/badges/shields/pull/5346#issuecomment-661238002
https://github.com/badges/shields/pull/5346#issuecomment-662007456

Reply:
any reason not to just ignore it with `// lgtm [js/hardcoded-credentials]` on that line? unsure why we'd want a less optimal solution to impress a linter

Reply:
> unsure why we'd want a less optimal solution to impress a linter

Personally I think it's important for our project to not have screaming "hardcoded credential" errors that are displayed as part of the analysis of our project; it's not a matter of "impressing" a linter. Such warnings are legitimate causes of concerns for our users, especially if they don't have the context/bandwidth to dig into the causes of warnings.

That being said, I wasn't aware that lgtm supported inline disables (thanks for the tip!), and would be quite content to leverage that with the original solution and some inline explanatory comments.

Reply:
Thanks for the feedback. Will revert the last commit and add an LGTM exclusion instead.

Note to myself for a future PR: we have some existing false positives which could benefit from this as well.

Reply:
You're gonna laugh.

According to the [LGTM documentation](https://help.semmle.com/lgtm-enterprise/user/help/alert-suppression.html), "suppression comments only apply to the line they are on". So I shoved the `// lgtm [js/hardcoded-credentials]` on the same line.

Then I ran Prettier. It didn't like what I did, so it put the comment on the next line. Same for the shortened catch all `// lgtm` equivalent. So I had to add a `// prettier-ignore`  exclusion as well to the whole method.

Here's what I've ended up with:
```javascript
// prettier-ignore
withBearerAuthHeader(requestParams, bearerKey = 'Bearer') { // lgtm [js/hardcoded-credentials]
  return this._withAnyAuth(requestParams, requestParams =>
    this.constructor._mergeHeaders(
      requestParams,
      this._bearerAuthHeader(bearerKey)
    )
  )
}
```

Not the end of the world, but I'm not impressed by what I ended up with either. It's been a long day on my side, will shut down the IDE for now and sleep over this.

Reply:
Well, I did come up with something that avoided the Prettier ignore, but the LGTM suppression doesn't seem to work. Anything obvious I've done wrong?

Reply:
@s0 Thought I'd mention you: we're having difficulty placing a lgtm ignore on the same line as the problem due to conflicts with Prettier. I'm curious if you have encountered this elsewhere!

Reply:
Hehe, this is a funny problem. Reminds me of prettier/eslint fix loops!

@paulmelnikow I'm no longer working for Semmle/GitHub, but I think I can shed some light on this. What you have there is probably fine and will unlikely introduce an alert, if you look at the suppression instructions, it says:

> If you make changes to the suppression of alerts in a pull request, please note that theses changes will only be effective once that pull request is merged. See Alert suppression for more details.

I think the reasoning behind this was to prevent people hiding new problems at review time by also adding comments?

cc @chrisgavin ?

Reply:
Thanks for the info @s0! As a small improvement suggestion, I feel it would be less confusing if the alert explicitly specified "this alert will be suppressed once the PR is merged", at least it proves that the suppression comment was well-formed. :)

Reply:
> @paulmelnikow I'm no longer working for Semmle/GitHub

I didn't realize this @s0! Thanks for the pointers!